draft-ietf-oauth-v2-1.md

```diff
@@ -1062,31 +1042,200 @@ request using a pre-defined default value or fail the request
 indicating an invalid scope.  The authorization server SHOULD
 document its scope requirements and default value (if defined).
 
+## Token Response {#token-response}
 
-# Obtaining Authorization {#obtaining-authorization}
+If the access token request is valid and authorized, the
+authorization server issues an access token and optional refresh
+token. 
 
-To request an access token, the client obtains authorization from the
-resource owner. OAuth defines two authorization grant types: authorization code
-and client credentials.  It also provides an extension mechanism for defining additional grant types.
+If the request client
+authentication failed or is invalid, the authorization server returns
+an error response as described in {{token-error-response}}.
 
+The authorization server issues an access token and optional refresh
+token by creating an HTTP response body using the `application/json`
+media type as defined by {{RFC8259}} with the following parameters
+and an HTTP 200 (OK) status code:
 
-## Authorization Code Grant {#authorization-code-grant}
+"access_token":
+:    REQUIRED.  The access token issued by the authorization server.
 
-The authorization code grant type is used to obtain both access
-tokens and refresh tokens.
+"token_type":
+:    REQUIRED.  The type of the access token issued as described in
+     {{access-token-types}}.  Value is case insensitive.
 
-Since this is a redirect-based flow, the client must be capable of
-initiating the flow with the resource owner's user agent (typically a web
-browser) and capable of being redirected back to from the authorization server.
+"expires_in":
+:    RECOMMENDED.  The lifetime in seconds of the access token.  For
+     example, the value `3600` denotes that the access token will
+     expire in one hour from the time the response was generated.
+     If omitted, the authorization server SHOULD provide the
+     expiration time via other means or document the default value.
 
-~~~~~~~~~~
-+----------+
-| Resource |
-|   Owner  |
-|          |
-+----------+
-     ^
-     |
+"refresh_token":
+:    OPTIONAL.  The refresh token, which can be used to obtain new
+     access tokens using the same authorization grant as described
```

Comment:
This sounds weird here now, how about "...using the refresh token grant as described..."

Reply:
good catch! I changed the text to refer to the grant specified in the token request